Cabal/doc/developing-packages.rst

```diff
@@ -1735,12 +1741,20 @@ system-dependent values for these fields.
     an executable in the package will override any of the hard-coded packages with the same name,
     and a hard-coded package will override any executable on the ``PATH``.
 
-    In the first two cases, the list entry is desguared into a :pkg-field:`build-tool-depends` entry.
+    In the first two cases, the list entry is desugared into a :pkg-field:`build-tool-depends` entry.
     In the first case, the entry is desugared into a :pkg-field:`build-tool-depends` entry by prefixing with ``$pkg:``.
     In the second case, it is desugared by looking up the package and executable name in a hard-coded table.
     In either case, the optional version bound is passed through unchanged.
     Refer to the documentation for :pkg-field:`build-tool-depends` to understand the desugared field's meaning, along with restrictions on version bounds.
 
+    If backwards-compatibility is important, you may opt to use
+    :pkg-field:`build-tool-depends` for cases (1) and (2), as they
+    are supported by old versions of Cabal.  Case (3) should not
+    be relied upon, as support for taking executables from ``PATH``
```

Comment:
Are you referring to the PATH feature? Unfortunately I can't really remove it, because it's legal to put an internal exe in build-tools, and when I implemented per-component build I needed this to keep working.